frontend/Container.js

```diff
@@ -37,14 +46,51 @@ class Container extends React.Component {
     },
     extraTabs: {[key: string]: () => React$Element},
   };
+  state: State;
+
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      isVertical: (window.innerWidth < IS_VERTICAL_BREAKPOINT),
+    };
+  }
+
+  componentDidMount() {
+    window.addEventListener('resize', this.handleResize, false);
+    this.setState({
+      isVertical: (window.innerWidth < IS_VERTICAL_BREAKPOINT),
+    });
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.handleResize);
+    clearTimeout(resizeTimeout);
+  }
+
+  handleResize = (e: Event) => {
+    if (!resizeTimeout) {
+      resizeTimeout = setTimeout(this.handleResizeTimeout, 50);
+    }
+  }
```

Comment:
I think you might need to add semicolon at the end of property initializers. This probably breaks some the (potentially old) Flow version we're using.

```js
handleStuff = () => {
  // ...
};

handleOtherStuff = () => {
  // ...
};
```

You don't need to do it for regular methods though.